pallets/ds-maps/src/tests.rs

```diff
@@ -114,41 +117,248 @@ fn it_try_add_root_unauthorized() {
+                REGISTRAR_1_ACCOUNT_ID,
+                super::REGISTRAR_ROLE
+          ));
+          let raw_coords: [i32; 6] = [
+              465587600,
+              312529919,
+              8388608,
+              469815744,
+              318558719,
+              16777216
+          ];
+          let delta: i32 = 838860;
+          assert_noop!(
+            DSMapsModule::raw_root_add(
+                Origin::signed(REGISTRAR_1_ACCOUNT_ID),
+                raw_coords,
+                delta
+            ),
+            Error::PageLimitExceeded
+          );
+
+          let root = DSMapsModule::root_box_data(ROOT_ID);
+          assert!(!root.is_active());
+      });
+}
+
+#[test]
+fn it_tries_to_add_too_big_root() {
+    new_test_ext().execute_with(|| {
+        assert_ok!(
+            DSAccountsModule::account_add(
+                Origin::signed(ADMIN_ACCOUNT_ID),
+                REGISTRAR_1_ACCOUNT_ID,
+                super::REGISTRAR_ROLE
+        ));
+        assert_noop!(
+            DSMapsModule::root_add(
+                Origin::signed(REGISTRAR_1_ACCOUNT_ID),
+                construct_custom_box("0.0", "0.0", "250.0", "250.0",),
+                coord(DELTA),
+            ),
+            Error::PageLimitExceeded
+        );
+        assert_noop!(
+            DSMapsModule::root_add(
+                Origin::signed(REGISTRAR_1_ACCOUNT_ID),
+                construct_custom_box("0.0", "0.0", "45.1", "50.9"),
+                coord(DELTA),
+            ),
+            Error::PageLimitExceeded
+        );
+    });
+}
+
+#[test]
+fn it_tries_to_add_root_with_incorrect_coordinates() {
+    new_test_ext().execute_with(|| {
+        assert_ok!(
+            DSAccountsModule::account_add(
+                Origin::signed(ADMIN_ACCOUNT_ID),
+                REGISTRAR_1_ACCOUNT_ID,
+                super::REGISTRAR_ROLE
+        ));
+        assert_noop!(
+            DSMapsModule::root_add(
+                Origin::signed(REGISTRAR_1_ACCOUNT_ID),
+                construct_custom_box("250.0", "0.0", "0.0", "250.0"),
+                coord(DELTA),
+            ),
+            Error::InvalidCoords
+        );
+        assert_noop!(
+            DSMapsModule::root_add(
+                Origin::signed(REGISTRAR_1_ACCOUNT_ID),
+                construct_custom_box("100.0", "50.9", "0.1", "0.0"),
+                coord(DELTA),
+            ),
+            Error::InvalidCoords
+        );
+    });
+}
+
+// 4 pages as square (2x2)
+// +++++++++
+// |___+___|
+// |___+___|
+// |+++++++|
+// |___+___|
+// |___+___|
+// +++++++++
+#[test]
+fn it_tries_to_add_root_as_square_2x2() {
+    new_test_ext().execute_with(|| {
+        assert_ok!(
+            DSAccountsModule::account_add(
+                Origin::signed(ADMIN_ACCOUNT_ID),
+                REGISTRAR_1_ACCOUNT_ID,
+                super::REGISTRAR_ROLE
+        ));
+
+        let bounding_box = construct_custom_box("0.051", "0.0", "0.5", "0.75", );
```

Comment:
"0.051" is because of I10F22 type features. If I write "0.05", it will be represented as "0.049", which is not allowed under the current implementation